### DISCUSSION TITLE
using custom background bg-[#......] doesn't work with dynamic color code value

### DISCUSSION THREAD

**Discussion by zwelhtetyan** [tailwindlabs/tailwindcss#10017] [Help] [answered]
`["tags": [
      {
        "tagName": "flowers",
        "color": "#ffd615",
        "_id": "6390b759e1d8e3c5248a4aa7"
      },
      {
        "tagName": "nature",
        "color": "#2cb978",
        "_id": "6390b759e1d8e3c5248a4aa8"
      }]`

![Screenshot from 2022-12-08 11-12-20](https://user-images.githubusercontent.com/89287841/206358760-8252dd3c-6456-4d65-980b-cfdac71590fe.png)

![Screenshot from 2022-12-08 11-13-48](https://user-images.githubusercontent.com/89287841/206358816-13798865-0274-4214-8e1d-506fc9fed6fd.png)

![Screenshot from 2022-12-08 11-14-31](https://user-images.githubusercontent.com/89287841/206358950-3bb74ce1-bb8a-4ad9-99de-4d4e5f5f12f1.png)

**The very first tag is work and others are not.**

**Comment by andymccray** [comment, marked as answer]
[Per the docs](https://tailwindcss.com/docs/content-configuration#dynamic-class-names), Tailwind will `only find classes that exist as complete unbroken strings in your source files`. Those colours aren’t being picked up because you're concatenating the `bg` prefix and the dynamic `color` variable.

To get around this you can pass in the _complete unbroken string_:

```
["tags": [ { "tagName": "flowers", "color": "bg-[#ffd615]", "_id": "6390b759e1d8e3c5248a4aa7" }, { "tagName": "nature", "color": "bg-[#2cb978]", "_id": "6390b759e1d8e3c5248a4aa8" }]
```

and then return the `color` variable:

```js
return <div className={`${color} px-2 rounded-md`}>{tagName}</div>
```
